content/chapter-destinations/destination-s3/_index.md

```diff
@@ -0,0 +1,205 @@
+
+## Options
+
+The following options are specific to the `s3()` destination.
+
+<!-- FIXME are there any common/generic/inherited options that we should reference? -->
+
+## access-key()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The `ACCESS_KEY` of the service account used to access the S3 bucket. (Together with [`secret-key()`](#secret-key).)
+
+## bucket()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The name of the S3 bucket, for example, `my-bucket`
+
+## canned-acl()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `empty` |
+
+*Description:* The ACL assigned to the object, if specified, for example, `bucket-owner-read`. The following values are valid:
+
+`authenticated-read`, `aws-exec-read`, `bucket-owner-full-control`, `bucket-owner-read`, `log-delivery-write`, `private`, `public-read`, `public-read-write`
+
+If you configure an invalid value, the default is used.
+
+## chunk-size()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `5MiB` |
+
+*Description:* The size of log messages that  {{% param "product_name" %}} writes to the S3 object in a batch.
+<!-- FIXME how does is relate to compression? -->
+
+## compression()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | boolean |
+| Default: | `no` |
+
+*Description:* <!-- FIXME -->
+
+## compresslevel()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer |
+| Default: | `9` |
+
+*Description:* <!-- FIXME -->
+
+## flush-grace-period()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer (seconds) |
+| Default: | `60` |
+
+*Description:* After the grace period expires and no new messages are routed to the destination, {{% param "product_name" %}} flushes the contents of the buffer to the S3 object even if the volume of the messages in the buffer is lower than [`chunk-size()`](#chunk-size).
+
+## max-object-size()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `5120GiB` |
+
+*Description:* The maximal size of the S3 object.
+<!-- What happens if we reach it, we start a new object? -->
+
+## max-pending-uploads()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer |
+| Default: | `32` |
+
+*Description:* <!-- FIXME -->
```

Comment:
The `max-pending-uploads()` and `upload-threads()` options are available to configure the upload of the chunks.
Uploading happens in multiple threads to minimize the effect of the network overhead. `upload-threads()` limits the maximum number of parallell uploads. `max-pending-uploads()` limits the number of chunks that are waiting (in the work queue of the upload threads) to get uploaded.